pkg/local/unix_permissions.go

```diff
@@ -0,0 +1,75 @@
+package local
+
+import (
+	"encoding/json"
+	"os"
+	"strings"
+	"syscall"
+
+	"github.com/treeverse/lakefs/pkg/api/apigen"
+	"github.com/treeverse/lakefs/pkg/api/apiutil"
+	"github.com/treeverse/lakefs/pkg/uri"
+)
+
+const (
+	DefaultFileMask            = 0o100666
+	UnixPermissionsMetadataKey = apiutil.LakeFSMetadataPrefix + "unix-permissions"
+)
+
+var (
+	// umask - internal, init only once. Use only via GetUmask call
+	umask = -1
+	// defaultOwnership - internal, init only once. Use only via getDefaultPermissions call
+	defaultOwnership *unixOwnership
+)
+
+type unixOwnership struct {
+	UID int
+	GID int
+}
+
+type UnixPermissions struct {
+	unixOwnership
+	Mode os.FileMode
+}
+
+func GetUmask() int {
+	if umask < 0 {
+		umask = syscall.Umask(0)
+		syscall.Umask(umask)
+	}
+	return umask
+}
+
+func getDefaultPermissions(isDir bool) UnixPermissions {
+	mode := DefaultFileMask - GetUmask()
+	if isDir {
+		mode = DefaultDirectoryMask - GetUmask()
+	}
+	if defaultOwnership == nil {
+		defaultOwnership = &unixOwnership{
+			UID: os.Getuid(),
+			GID: os.Getgid(),
+		}
+	}
+	return UnixPermissions{
+		unixOwnership: *defaultOwnership,
+		Mode:          os.FileMode(mode),
+	}
+}
+
+// getUnixPermissionFromStats - Get unix mode and ownership from object metadata, fallback to default permissions in case metadata doesn't exist
+func getUnixPermissionFromStats(stats apigen.ObjectStats) (*UnixPermissions, error) {
```

Comment:
`GetUnixPermissionFromStats`?

Reply:
No need to make it public (at least for now)